tests/console/pam.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This PM is to run a suite of tests about PAM, and gets a TAP
+#          format result. Four files(pam.sh, pam_test, pam_test.sh,
+#          system-default) are included. pam.sh is a "bats" script, and
+#          includes 25 tests for PAM. The other files are from the project
+#          "pam_test" which can be used to test a PAM stack for authentication
+#          and password change. The link: https://github.com/pbrezina/pam-test
+#   Steps:
+#       - add qa-head repo and install the tool "bats" from qa-head repo.
+#         Bats is a TAP-compliant testing framework for Bash.
+#       - download "pam_test" dir(automated tests of PAM) from "data" dir
+#         on openqa host if "prepare_test_data" perl module desn't run.
+#       - run the suite of tests in SUT.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+sub run {
+    select_console 'root-console';
+
+    my $version      = get_required_var('VERSION');
+    my $qa_head_repo = is_sle() ? "http://download.suse.de/ibs/QA:/Head/SLE-$version/" : 'https://download.opensuse.org/tumbleweed/repo/oss/';
+    zypper_ar("$qa_head_repo", name => 'qa-head-repo');
+    zypper_call('install bats pam-test pam pam-config snapper');
```

Comment:
On a side note, pam-test is only available for tumbleweed. Please ensure that your test is only scheduled for that scenario.

Reply:
Yes, pam-test is available for tumbleweed, and I don’t have permission to schedule my test for any scenario, so I don’t know if it is available for other scenarios.
But I asked @plusky (because he helped me build pam-test package), he told me that pam-test was added to openSUSE:Factory and no need for QA:Head, and the link "https://download.opensuse.org/tumbleweed/repo/oss/" is exactly openSUSE:Factory, although the link looks like a tumbleweed repo.

Reply:
Sorry, I meant distribution, not scenario :).

Reply:
I see. I am try to add pam-test into devel:openSUSE:QA project for supporting tumbleweed and Leap 15.0/15.1.